pkg/middleware/validation.go

```diff
@@ -35,7 +36,9 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 					return
 				}
 
-				if !valid {
+				maybeLogValidationReason(valid, r, mod, version)
+
+				if valid.Status != http.StatusOK {
```

Comment:
Given the previous comment, we can then just check `if valid.Valid` -- or maybe rename the variable and it will read better: `if response.Valid`

Reply:
https://github.com/gomods/athens/pull/1517/commits/12dba10897b22c735db29b02d50fd64a85bba2ea